grakn-graph/src/main/java/ai/grakn/graph/internal/ResourceImpl.java

```diff
@@ -71,9 +73,15 @@
      */
     @Override
     public Collection<Thing> ownerInstances() {
-        return getShortcutNeighbours().stream().
+        //Get Owner via implicit structure
+        Set<Thing> owners = getShortcutNeighbours().stream().
```

Comment:
If `getShortcutNeighbours` also followed relation edges, you might be able to simplify this and some other cases.